### ISSUE TITLE
How to get list of Loyal Followers?

### ISSUE THREAD

**Issue by lazlazlaz** [InstaPy/InstaPy#4487]
I use quickstart.py to simply unfollow non-followers.

After the script has finished in the end notes I got a `Found 602 loyal followers!  ~will not unfollow them` message. 

This made me think. I could just unfollow everyone but these accounts.

Where do I find this list? It must exist somewhere, right?

**Comment by synchronizing** [comment]
~~I believe that would be inside the autogenerated `InstaPy` folder: `InstaPy/logs/{account_name}/followRestriction.json`.~~

Edit: Real explanation below.

**Comment by lazlazlaz** [comment]
Thanks, but that file is empty for me...

**Comment by synchronizing** [comment]
Inside [`unfollow_util.py`](https://github.com/timgrossmann/InstaPy/blob/138bd155b171f6efc7d259707616aea149e63521/instapy/unfollow_util.py#L305) it seems that `loyal_users` is defined as the following:

```python
 loyal_users = [user for user in unfollow_list if user in all_followers]
```

Where `all_followers` is all your current followers. Therefore, from my understanding, loyal followers are actually those who follow you back -- InstaPy does not unfollow them and considers them "loyal".

In other words, the intersection of your following and followers list is your loyal followers. 

---

You can use [`grab_followers()`](https://github.com/timgrossmann/InstaPy/blob/master/DOCUMENTATION.md#grab-followers-of-a-user) and [`grab_following()`](https://github.com/timgrossmann/InstaPy/blob/master/DOCUMENTATION.md#grab-following-of-a-user) on your own account, convert them both into [Python's set](https://docs.python.org/3.7/library/stdtypes.html#set-types-set-frozenset), and then simply get their intersection; `loyal_followers = account_following & account_followers`.

**Comment by lazlazlaz** [comment]
That's weird. The loyal followers are only around 600 whereas if I unfollow all non-followers I still have around 2000 accounts I'm following.

**Comment by synchronizing** [comment]
If you notice inside the [`unfollow_util.py`](https://github.com/timgrossmann/InstaPy/blob/138bd155b171f6efc7d259707616aea149e63521/instapy/unfollow_util.py#L275) file, the definition for `unfollow_list` is the following:

```python
if customList is True:
            logger.info("Unfollowing from the list of pre-defined usernames\n")
            unfollow_list = customList_data
elif InstapyFollowed is True:
            logger.info("Unfollowing the users followed by InstaPy\n")
            unfollow_list = list(automatedFollowedPool["eligible"].keys())       
elif nonFollowers is True:
            logger.info("Unfollowing the users who do not follow back\n")
            """  Unfollow only the users who do not follow you back """
            unfollow_list = get_nonfollowers(browser, username, relationship_data, False, True, logger, logfolder)
```

So what is going on is that the list InstaPy is getting for `unfollow_list` is a _subset_ of your following list. This means that the 600 loyal followers account that you are getting is a _subset_ of your total loyal follower's account. 

In other words, depending on your configuration, InstaPy is retrieving a smaller list of your followers to unfollow. Therefore, when it checks that list (the smaller list of your followers) for loyal followers, it finds a smaller list of your total loyal followers -- explaining why you got 600 on that specific run, and not 2000 as expected. If you run InstaPy once again I would believe that the number would no longer be 600, and either more or less depending on other InstaPy mechanisms that I haven't looked into. 

The technique I posted above to get your loyal followers should still work with the above logic. 👍
